src/commonMain/kotlin/arrow.exact/Exact.kt

```diff
@@ -0,0 +1,23 @@
+package arrow.exact
+
+import arrow.core.*
+
+interface Exact<A, out B> {
+
+  fun from(value: A): Either<ExactError, B>
+
+  fun fromOrNull(value: A): B? {
+    return from(value).getOrNull()
+  }
+
+  fun fromOrThrow(value: A): B {
+    return when (val result = from(value)) {
+      is Either.Left -> throw ExactException(result.value.message)
+      is Either.Right -> result.value
+    }
+  }
+}
+
+open class ExactError(val message: String)
```

Comment:
Why an open class and not an interface?

Reply:
@CLOVIS-AI That's outdated. Have a look at #8 👈 and #13 👈

Reply:
Thanks for the link :)